ebl/src/test/java/org/dcsa/conformance/standards/ebl/checks/EBLChecksTest.java

```diff
@@ -291,4 +292,26 @@ void testValidRequestedCarrierClauses() {
     Set<String> invalidErrors = VALID_REQUESTED_CARRIER_CLAUSES.validate(rootNode);
     assertEquals(1, invalidErrors.size());
   }
+
+  @Test
+  void testValidConsignmentItemsReferenceTypes(){
+    ObjectNode rootNode = getRootNode();
+    ArrayNode consignmentItems = rootNode.putArray("consignmentItems");
+    ObjectNode consignmentItem = consignmentItems.addObject();
+    ArrayNode references = consignmentItem.putArray("references");
+    references.addObject().put("type", "CR");
+    references.addObject().put("type", "SPO");
+
+    Set<String> errors = VALID_CONSIGMENT_ITEMS_REFERENCE_TYPES.validate(rootNode);
+    assertEquals(0, errors.size());
+
+    references.addObject().put("type", "CRR");
+    Set<String> invalidErrors = VALID_CONSIGMENT_ITEMS_REFERENCE_TYPES.validate(rootNode);
+    assertEquals(1, invalidErrors.size());
+
+  }
+
+  private ObjectNode getRootNode() {
+    return objectMapper.createObjectNode();
+  }
```

Comment:
Advise: You can also make this a field of the class:
`private final ObjectNode rootNode = objectMapper.createObjectNode();`
So you can remove all the lines with: `ObjectNode rootNode = getRootNode();` and this method as well.